dot/network/block_announce.go

```diff
@@ -214,26 +214,62 @@ func (s *Service) validateBlockAnnounceHandshake(peer peer.ID, hs Handshake) err
 
 	bestBlockNum := big.NewInt(int64(bhs.BestBlockNumber))
 
+	np, ok := s.notificationsProtocols[BlockAnnounceMsgType]
+	if !ok {
+		err = s.RegisterNotificationsProtocol(
+			blockAnnounceID,
+			BlockAnnounceMsgType,
+			s.getBlockAnnounceHandshake,
+			decodeBlockAnnounceHandshake,
+			s.validateBlockAnnounceHandshake,
+			decodeBlockAnnounceMessage,
+			s.handleBlockAnnounceMessage,
+		)
+		if err != nil {
+			return errors.New("failed to register blockannounce protocol")
+		}
+	}
+
+	np.mapMu.Lock()
+	defer np.mapMu.Unlock()
+
+	_, ok = np.handshakeData[peer]
+	if !ok {
+		logger.Error("peer handshake data is nil")
+		np.handshakeData[peer] = &handshakeData{
+			validated:      false,
+			received:       true,
+			responseSentCh: make(chan struct{}),
```

Comment:
just return an error here too, it should always have been set (see notifications.go `createNotificationsMessageHandler`)